internal/index/index.go

```diff
@@ -76,6 +84,16 @@ func New(ctx context.Context, repo Lister, ignorePacks restic.IDSet, p *restic.P
 				return nil
 			}
 
+			if res, ok := oldIdx.Packs[id]; ok {
```

Comment:
This looks like a race condition with the `for res := range outputCh` loop. Two coroutines are working on `idx` and I don't see any synchronization in the AddPack method.

Replacing `repo.ListPack` with a lookup into the old index should avoid that race condition.

Reply:
Or better: Just send the pack from the oldIdx over the `outputCh` channel